### ISSUE TITLE
JIT for POWER?

### ISSUE THREAD

**Issue by madscientist159** [ptitSeb/box64#242]
Just tried this with the UT2004 retail binary on a ppc64le box, and the main menu came up without any notable issues.  It's **extremely** slow and quite stuttery (far worse than QEMU), but I'm assuming that's mostly because of the missing JIT for the ppc64le architecture,

Would you be interested in porting the JIT to ppc64le?  It's the only open ISA with high performance CPUs on the market, and we'd be happy to provide a free development box in return for the effort.

Thank you!

**Comment by ptitSeb** [comment]
So, RISCV64 Dynarec is in a good enough state now.

**Comment by JeremyRand** [comment]
What would be needed at this point to move this forward? You mentioned hardware access -- what would be the obstacles to doing this with remote access?

**Comment by TMinusBlastedRocket** [comment]
I have an IBM S824 (POWER8) I can give you root bare metal on

**Comment by ptitSeb** [comment]
> I have an IBM S824 (POWER8) I can give you root bare metal on 24/7 (power company and internet outages excluded, im a random on the interwebs not a miracle worker)

I'm a bit busy for now, but lets talk about that beggining of next year, and see if we can make it work.

**Comment by TMinusBlastedRocket** [comment]
> > I have an IBM S824 (POWER8) I can give you root bare metal on 24/7 (power company and internet outages excluded, im a random on the interwebs not a miracle worker)
> 
> I'm a bit busy for now, but lets talk about that beggining of next year, and see if we can make it work.

Cool

**Comment by TMinusBlastedRocket** [comment]
> > I have an IBM S824 (POWER8) I can give you root bare metal on 24/7 (power company and internet outages excluded, im a random on the interwebs not a miracle worker)
> 
> I'm a bit busy for now, but lets talk about that beggining of next year, and see if we can make it work.

You still wanna do this?

**Comment by ptitSeb** [comment]
> > > I have an IBM S824 (POWER8) I can give you root bare metal on 24/7 (power company and internet outages excluded, im a random on the interwebs not a miracle worker)
> > 
> > 
> > I'm a bit busy for now, but lets talk about that beggining of next year, and see if we can make it work.
> 
> You still wanna do this?

Not for the next 3 months I afraid, I'm still too busy to take on this in a reasonable way and timing.

**Comment by madscientist159** [comment]
----- Original Message -----
> From: "ptitSeb" ***@***.***>
> To: "ptitSeb/box64" ***@***.***>
> Cc: "Timothy Pearson" ***@***.***>, "Author" ***@***.***>
> Sent: Friday, February 16, 2024 12:09:25 PM
> Subject: Re: [ptitSeb/box64] JIT for POWER? (Issue #242)

>> > > I have an IBM S824 (POWER8) I can give you root bare metal on 24/7 (power
>> > > company and internet outages excluded, im a random on the interwebs not a
>> > > miracle worker)
>> > 
>> > 
>> > I'm a bit busy for now, but lets talk about that beggining of next year, and see
>> > if we can make it work.
>> 
>> You still wanna do this?
> 
> Not for the next 3 months I afraid, I'm still too busy to take on this in a
> reasonable way and timing.

When ready we can also offer POWER9 access, which has new instructions vs. POWER8 that could be of assistance for a JIT.

**Comment by runlevel5** [comment]
It is always great to see responses Tim of RaptorCS. My offer to gain access to my machine also stand, I could setup RDP or whatever prefer way to get the dev team started

**Comment by ptitSeb** [comment]
> It is always great to see responses Tim of RaptorCS. My offer to gain access to my machine also stand, I could setup RDP or whatever prefer way to get the dev team started

Box64 really need Little Endian CPU, and I don't think that an option on Amiga.

Having Box64 running on Big Endian CPU would require extensive work on the Wrapping side, probably more extensive then what Box32 already require....

**Comment by bobpaul** [comment]
> Box64 really need Little Endian CPU, and I don't think that an option on Amiga.


That's fine. Both Power8 and Power9 both can boot both big endian and little endian kernels. ppc64le with 4k pages is a common for desktop Linux on Power systems. I know @runlevel5 is active on the RaptorCS forums; I'm certain he's offering access to a Power9 ppc64le system, not an Amiga.

**Comment by SteffenHaeuser** [comment]
I was mislead by google to the wrong project's Issue section. That's why I removed my comment again when I realized I ended up on box64 issue section;-) Thx for the info anyways.

**Comment by runlevel5** [comment]
> That's fine. Both Power8 and Power9 both can boot both big endian and little endian kernels. ppc64le with 4k pages is a common for desktop Linux on Power systems. I know @runlevel5 is active on the RaptorCS forums; I'm certain he's offering access to a Power9 ppc64le system, not an Amiga.

Non-UI low-level I believe could be no issue however GUI could be problematic as remote desktop can be quite slow. Having said that I am more than happy to offer access to my box if it helps the porting efforts
